sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncReadBenchmark.java

```diff
@@ -48,24 +48,26 @@ protected void hookOnError(Throwable throwable) {
 
     AsyncReadBenchmark(Configuration cfg) {
         super(cfg);
+        // TODO: once all benchmarks move to v4 api, we should rely on the container which is read in the parent, and remove this.
+        cosmosAsyncContainer = v4Client.getDatabase(cfg.getDatabaseId()).getContainer(cfg.getCollectionId()).read().block().getContainer();
     }
 
     @Override
-    protected void performWorkload(BaseSubscriber<ResourceResponse<Document>> baseSubscriber, long i) throws InterruptedException {
+    protected void performWorkload(BaseSubscriber<CosmosAsyncItemResponse> baseSubscriber, long i) throws InterruptedException {
         int index = (int) (i % docsToRead.size());
-        RequestOptions options = new RequestOptions();
-        options.setPartitionKey(new PartitionKey(docsToRead.get(index).getId()));
+        Document doc = docsToRead.get(index);
 
-        Flux<ResourceResponse<Document>> obs = client.readDocument(getDocumentLink(docsToRead.get(index)), options);
+        String partitionKeyValue = doc.getId();
+        Mono<CosmosAsyncItemResponse> result = cosmosAsyncContainer.getItem(doc.getId(), partitionKeyValue).read();
 
         concurrencyControlSemaphore.acquire();
```

Comment:
Sync vs async we need to baseline its impact.

Reply:
sure on my list.